Library/Homebrew/command.rb

```diff
@@ -0,0 +1,156 @@
+module Homebrew
+  class Command
+    class << self
+      attr_reader :command_name, :valid_options, :description, :help_output, :man_output
+    end
+
+    def self.initialize
+      @valid_options = []
+      @description = nil
+      @command_name = nil
+      @help_output = nil
+      @man_output = nil
+      @root_options = []
+    end
+
+    def self.command(cmd)
+      @command_name = cmd
+    end
+
+    def self.options(&block)
+      initialize
+      @parent = nil
+      class_eval(&block)
+      generate_help_and_manpage_output
+      build_methods_from_options
+    end
+
+    def self.build_methods_from_options
+      @valid_options.each do |hash|
+        option_name = hash[:option]
+        option_name = option_name.gsub(/^--/, "").tr("-", "_")
+        Command.define_singleton_method("#{option_name}?") do
+          ARGV.include? "--#{option_name.tr("_", "-")}"
+        end
+      end
+    end
+
+    def self.add_valid_option(option, desc)
+      valid_option = { option: option, desc: desc, child_options: nil }
+      @valid_options.push(valid_option)
+    end
+
+    def self.option(key, value, &block)
```

Comment:
@woodruffw, we can check the ```key``` variable here on line 43. And if that variable contains a ```=```, then we can assign that value for that key. We would also have to add another attribute to the hash which will indicate the value (e.g ```gcc```) for an option(e.g. ```--cc```). What do you think?